**<p style="font-size: 35px; text-align: center">Confidence Intervals Estimation</p>**



***<center>Miguel Ángel Vélez Guerra</center>***

<hr/>

![intervals](http://sphweb.bumc.bu.edu/otlt/MPH-Modules/QuantCore/PH717_ConfidenceIntervals-OneSample/Wordle-ConfidenceIntervals.png)





<hr />
<hr />

**<p id="tocheading">Tabla de contenido</p>**
<br/>
<div id="toc"></div>


In [24]:
%%javascript
$.getScript('../resources/table_of_contents.js')

<IPython.core.display.Javascript object>

<hr />
<hr />

## Imports

In [2]:
#-------Importing from other folder------#

import sys
sys.path.insert(0, "../resources/")

import mstats

#-----------Miguel's statistics----------#

In [3]:
import scipy.stats as ss
import numpy as np

<hr />
<hr />

## 1.  Intervalos de confianza para la media poblacional con muestras grandes
Un teatro de cine local desea desarrollar un intervalo para estimar las cajas promedio de
palomitas de maíz que se venden por sala de cine. Si los registros llevados para 70 salas
revelan un promedio de 54.98 cajas y una desviación estándar de 12.7, calcule e interprete
un intervalo de confianza del 92% para la media poblacional.

In [4]:
n_cine = 70 # Tamaño de la muestra
x__cine = 54.98 # Promedio muestral
s_cine = 12.7 # Desviación estándar muestral
alpha_cine = 0.92 # Nivel de confianza

In [5]:
mstats.interval_norm(n_cine, x__cine, s_cine, alpha_cine)

(52.32256300052874, 57.637436999471255)

**R/** El teatro de cine local tiene un *92%* de confianza en que el promedio poblacional de cajas de palomitas de maíz que se venden por sala de cine, está entre **52.32256300052874** y **57.637436999471255** cajas.

<hr/>
<hr/>

## 2. Intervalos de confianza para la media poblacional con muestras grandes
Una muestra de 121 llamadas al número 900 que usted maneja tiene una duración
promedio de 16.6 minutos y una desviación estándar de 3.63 minutos. Usted pretende
discontinuar el servicio a menos que la duración promedio sea superior a 18 minutos. En
un nivel de confianza del 95%, ¿cuál es su decisión?

In [6]:
n_llamadas = 121 # Tamaño de la muestra
x__llamadas = 16.6 # Promedio muestral
s_llamadas = 3.63 # Desviación estándar muestral
alpha_llamadas = 0.95 # Nivel de confianza

In [7]:
mstats.interval_norm(n_llamadas, x__llamadas, s_llamadas, alpha_llamadas)

(15.953211885101783, 17.24678811489822)

**R/** Con un nivel de confianza del *95%* podemos asegurar que la duración promedio de llamadas al número 900 está entre **15.953211885101783** y **17.24678811489822** minutos.

Entonces, nuestra decisión será discontinuar el servicio, pues la duración promedio no parece que pueda superar los **18** minutos.

<hr />
<hr />

## 3. Intervalos de confianza para la media poblacional con muestras pequeñas
Greenleaf Lawn Care descubre que el costo promedio de adornar los jardines de 20 casas
del área es de 2365, con s = 983. Al nivel de confianza del 99%, ¿Qué costo promedio
estimaría usted para adornar los jardines de todas las casas del área?

In [8]:
n_jardines = 20 # Tamaño de la muestra
x__jardines = 2365 # Promedio muestral
s_jardines = 983 # Desviación estándar muestral
alpha_jardines = 0.99 # Nivel de confianza

In [9]:
mstats.interval_t(n_jardines, x__jardines, s_jardines, alpha_jardines)

(1736.1508893202845, 2993.8491106797155)

**R/** Con un nivel de confianza del *99%* , se podría estimar que el costo promedio para adornar los jardinas de todas las casas del área está entre **1736.1508893202845** y **2993.8491106797155**

<hr />
<hr />

## 4. Intervalos de confianza para la proporción poblacional
Si de los 1098 turistas norteamericanos 896 recimiendan su viaje a Irlanda a sus amigos,
¿qué porcentaje de todos los turistas norteamericanos harían lo mismo con un nivel del
99% de confianza?

In [10]:
n_turistas = 1098 # Tamaño de la muestra
p_turistas = 895/1098 # Proporción muestral
alpha_turistas = 0.99 # Nivel de confianza

In [11]:
mstats.interval_prop(n_turistas, p_turistas, alpha_turistas)

(0.7849416025416293, 0.8452951916295911)

**R/** Se puede asegurar con un nivel de confianza del *99%* que la proporción de todos los turistas norteamericanos recomienden su viaje a Irlanda a sus amigos está entre **0.7849416025416293** y **0.8452951916295911**

<hr />
<hr />

## 5. Determinación del tamaño de la muestra para la media poblacional
¿Qué tan grande se requiere que sea una muestra para que proporcione una estimación
del 90% del número promedio de graduados de las universidades de la nación con un
error de 2000 estudiantes si una muestra piloto reporta que s = 8659?

In [12]:
alpha_graduados = 0.9 # Nivel de confianza
s_graduados = 8659 # Desviación estándar de la muestra piloto
error_graduados = 2000 # Error tolerado

In [13]:
mstats.n_avg(alpha_graduados, s_graduados, error_graduados)

51.0

**R/** Se requiere tomar una muestra de **51** graduados de las universidades de la nación, para que de esta manera se pueda estimar el número promedio de graduados con un nivel de confianza del *90%* y teniendo en cuenta un error menor a *2000* estudiantes.

<hr />
<hr />

## 6. Determinación del tamaño de la muestra para la proporción poblacional
La división de créditos de un banco comercial grande desea estimar con un nivel de
confianza del 99% la proporción de sus créditos que están en mora. Si el ancho del
intervalo es del 7%, ¿Cuántos créditos deben revisarse? ¿cuál es el error tolerable?

In [14]:
alpha_creditos = 0.99 # Nivel de confianza
p_creditos = 0.5 # Valor esperado para la proporción poblacional
error_creditos = 0.07 # Error tolerado

In [15]:
mstats.n_prop(alpha_creditos, p_creditos, error_creditos)

339.0

**R/** El error tolerable es del **7%**, y con un nivel de confianza del *99%* el número total de créditos que se deberían revisar para estimar la proporción de créditos que están en mora es de **339** créditos.

<hr />
<hr />

## 7. Intervalos de confianza para la proporción poblacional
Un investigador de la Administración Federan de Aviación (FAA) fue mencionado en una
emisión de febrero de The Washington Post, en la que dijo que de los 112 accidentes de
aerolíneas, “73 involucraban algún tipo de problema estructural con la aeronave”. Si tales
cifras son representativas, ¿Cuál es el intervalo de confianza para la proporción de
accidentes que involucran tal defecto estructural? Sea α = 0.01.

In [16]:
n_accidentes = 112 # Tamaño de la muestra
p_accidentes = 73/112 # Proporción muestral
alpha_accidentes = 0.99 # Nivel de confianza

In [17]:
mstats.interval_prop(n_accidentes, p_accidentes, alpha_accidentes)

(0.5358322364439551, 0.7677391921274735)

**R/** Con un nivel de confianza del *99%* se puede asegurar que la proporción de accidentes que involucran algún tipo de problema estrutural con la aeronave está entre **0.5358322364439551** y **0.7677391921274735**.

<hr />
<hr />

## 8. Intervalos de confianza para la proporción poblacional
The Wall Street Journal informó los esfuerzos realizados por Nestlé, la compañía de
alimentos más grande del mundo, para introducir un nuevo producto. La gerencia decidió
utilizar el sector de Chicago como mercado de prueba. Si más del 30% de la personas
expresaban un deseo por el producto, ellos considerarían el comercializarlo en un sector
más amplio. De las 820 personas a quienes se les practicó la prueba, 215 expresaron una
reacción positiva. ¿Un intervalo de confianza del 90% para la proporción de todos los
clientes quienes prefieren el producto hace que la gerencia continúe con sus planes de
comercialización?

In [18]:
n_clientes = 820 # Tamaño de la muestra
p_clientes = 215/820 # Proporción muestral
alpha_clientes = 0.9 # Nivel de confianza

In [19]:
mstats.interval_prop(n_clientes, p_clientes, alpha_clientes)

(0.23693104732523565, 0.2874591965772034)

**R/** Con un nivel de confianza del *90%* se puede estimar que la proporción de personas que expresan un deseo por el producto de Nestle está entre **0.23693104732523565** y **0.2874591965772034**.

Con lo que podemos concluir que Nestle no va a considerar comercializar el producto en un sector más amplio, puesto que no parece que a más del *30%* de las personas les interese el nuevo producto.

<hr />
<hr />

## 9. Determinación del tamaño de la muestra para la media poblacional
El director de su división le pide, como analista de mercadeo recientemente contratado,
que estime las ventas semanales promedio. Le advierte que debe mantener el error
dentro de 100 y mantener un nivel de confianza del 90%. ¿Cuántas semanas de datos
debe recolectar si la desviación estándar es 750?

In [20]:
alpha_ventas = 0.9 # Nivel de confianza
s_ventas = 750 # Desviación estándar de la muestra piloto
error_ventas = 100 # Error tolerado

In [21]:
mstats.n_avg(alpha_ventas, s_ventas, error_ventas)

152.0

**R/** Debemos recolectar **152** semanas de datos para poder estimar con un nivel de confianza del *90%* y un error menor de *100* las ventas semanales promedio de la empresa.

<hr />
<hr />

## 10. Determinación del tamaño de la muestra para la proporción poblacional
La Asociación de Finanzas Estudiantiles en Faber Collage está planeando una “Feria
primaveral” en la cual intentan vender camisetas con su logo. El tesorero desea un
estimado de la proporción de estudiantes que comprarán una camiseta. El estimado debe
proporcionar un nivel de confianza del 90% y el error no debe exceder del 3%. ¿Qué tan
grande debe tomarse la muestra?

In [22]:
alpha_camisetas = 0.9 # Nivel de confianza
p_camisetas = 0.5 # Valor esperado para la proporción muestral
error_camisetas = 0.03 # Error tolerado

In [23]:
mstats.n_prop(alpha_camisetas, p_camisetas, error_camisetas)

752.0

**R/** La muestra debe ser de un tamaño de **752** estudiantes, para que con un nivel de confianza del *90%* y un error menor de *3%* se pueda estimar la proporción de estudiantes que comprarán la camiseta.